civis/io/_tables.py

```diff
@@ -224,7 +258,12 @@ def f(x):
                                .format(script_id))
     url = outputs[0]["path"]
     if use_pandas:
-        data = pd.read_csv(url, **kwargs)
+        sortkeys, py_types, date_cols = _redshift_to_py(client, table_id)
```

Comment:
You need to test here if `table_id` is `None`. If it is, then we should revert back to pandas type inference.